src/Symfony/Component/HttpClient/EventSource.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Component\EventDispatcher\EventDispatcher;
+use Symfony\Component\HttpClient\Exception\ClientException;
+use Symfony\Component\HttpClient\Exception\InvalidArgumentException;
+
+final class EventSource extends EventDispatcher
+{
+    // ready state
+    const CONNECTING = 0;
+    const OPEN = 1;
+    const CLOSED = 2;
+
+    // events
+    const ON_MESSAGE = 'eventsource:onmessage';
+    const ON_OPEN = 'eventsource:onopen';
+    const ON_ERROR = 'eventsource:onerror';
+
+    public $lastEventId = '';
+    public $readyState = self::CLOSED;
+
+    private $client;
+    private $reconnectionTime = 30;
+    private $shouldDisconnect = false;
+
+    public function __construct(HttpClientInterface $client = null)
+    {
+        $this->client = $client ?: HttpClient::create($configuration['default_options'] ?? []);
```

Comment:
I think `$configuration` variable might be not defined.

Reply:
also `??` instead of `?:`